src/client/src/rt-components/platform/adapters/openfin/openFin.ts

```diff
@@ -116,3 +125,77 @@ export default class OpenFin extends BasePlatformAdapter {
       } as any),
   }
 }
+
+async function appRestoreHandler(workspaceApp: any, store: any) {
+  const ofApp = await fin.Application.getCurrent()
+  const openWindows = await ofApp.getChildWindows()
+
+  const opened = workspaceApp.childWindows.map(async (win: any, index: any) => {
+    if (!openWindows.some(w => w.identity.name === win.name)) {
+      const config: WindowConfig = {
+        name: win.name,
+        url: win.url,
+        width: win.bounds.width,
+        height: win.bounds.height,
+      }
+      await openDesktopWindow(
+        config,
+        () => {
+          store.dispatch(
+            LayoutActions.updateContainerVisibilityAction({
+              name: win.name,
+              display: true,
+            }),
+          )
+        },
+        { defaultLeft: win.bounds.left, defaultTop: win.bounds.top },
+      )
+
+      // we need to 'remove' the child window from the main window
+      store.dispatch(
+        LayoutActions.updateContainerVisibilityAction({
+          name: win.name,
+          display: false,
+        }),
+      )
+    } else {
+      await positionWindow(win)
+    }
+  })
+
+  await Promise.all(opened)
+  return workspaceApp
+}
+
+async function positionWindow(win: any): Promise<void> {
+  console.error('dans positionWindow', win)
+  try {
+    const { isShowing, isTabbed } = win
+
+    const ofWin = await fin.Window.wrap(win)
+    await ofWin.setBounds(win.bounds)
+
+    if (isTabbed) {
+      await ofWin.show()
+      return
+    }
+
+    await ofWin.leaveGroup()
+
+    if (isShowing) {
+      if (win.state === 'normal') {
```

Comment:
Create an enum type for these strings.

Reply:
I can do it, but do we really need it? The state is defined this way in openFin `state?: 'normal'|'minimized'|'maximized';` so if we type something else it won't work...

Reply:
I think it is more a question of convention used throughout RT, we define enum types for constant string literal.

Reply:
ok - done